### ISSUE TITLE
LineChart first spot cannot be nullSpot

### ISSUE THREAD

**Issue by andreferreiravntrs** [imaNNeoFighT/fl_chart#659]
I am trying to set an early line chart, where my horizontal values can start as null if there is no data assigned to the initial months.

```
        spots: [
          FlSpot.nullSpot,
          FlSpot.nullSpot,
          FlSpot(2, 83),
          FlSpot(3, 83),
          FlSpot(4, 82.25),
          FlSpot(5, 80),
          FlSpot(6, 79),
          FlSpot.nullSpot,
          FlSpot.nullSpot,
          FlSpot.nullSpot,
          FlSpot(10, 79),
        ],
```

If the initial Spot is set to nullSpot i get the following error:
```
The following assertion was thrown during paint():
Offset argument contained a NaN value.
'dart:ui/painting.dart':
Failed assertion: line 52 pos 10: '<optimized out>'

The relevant error-causing widget
When the exception was thrown, this was the stack
#2      _offsetIsValid (dart:ui/painting.dart:52:10)
#3      Canvas.drawCircle (dart:ui/painting.dart:4204:12)
#4      FlDotCirclePainter.draw
package:fl_chart/…/line_chart/line_chart_data.dart:824
#5      CanvasWrapper.drawDot
package:fl_chart/…/utils/canvas_wrapper.dart:74
#6      LineChartPainter._drawDots
package:fl_chart/…/line_chart/line_chart_painter.dart:263

The following RenderObject was being processed when the exception was fired: RenderLineChart#6f675
RenderObject: RenderLineChart#6f675
    parentData: <none> (can use size)
    constraints: BoxConstraints(w=376.0, h=280.0)
    size: Size(376.0, 280.0)

```
Am i doing something wrong with the settings of the LineChart?

```
LineChartData(
    gridData: FlGridData(
      show: true,
      getDrawingHorizontalLine: (value) {
        return FlLine(
          color: ColorPalette.paleGrey,
          strokeWidth: 1.0,
        );
      },
    ),
    titlesData: FlTitlesData(
      show: true,
      bottomTitles: SideTitles(
        showTitles: true,
        reservedSize: 20,
        getTextStyles: (value) =>
            const TextStyle(color: ColorPalette.blueyGrey, fontSize: 8),
        getTitles: (value) {
          if (lengthType == LengthType.monthly) {
            switch (value.toInt()) {
              case 0:
                return 'JAN';
              case 1:
                return 'FEB';
              case 2:
                return 'MAR';
              case 3:
                return 'APR';
              case 4:
                return 'MAY';
              case 5:
                return 'JUNE';
              case 6:
                return 'JUL';
              case 7:
                return 'AUG';
              case 8:
                return 'SEP';
              case 9:
                return 'OCT';
              case 10:
                return 'NOV';
              case 11:
                return 'DEC';
              default:
                return '';
            }
          } else {
            switch (value.toInt()) {
              case 0:
                return '1';
              case 13:
                return '14';
              case 27:
                return '28';
              default:
                return '';
            }
          }
        },
        margin: 15,
      ),
      leftTitles: SideTitles(
        showTitles: true,
        getTextStyles: (value) => const TextStyle(
          color: ColorPalette.aquaMarine,
          fontSize: 10,
        ),
        getTitles: (value) {
          switch (value.toInt()) {
            case 0:
              return '0';
            case 25:
              return '25';
            case 50:
              return '50';
            case 75:
              return '75';
            case 100:
              return '100';
          }
          return '';
        },
        reservedSize: 13,
      ),
    ),
    borderData: FlBorderData(
        show: true, border: Border.all(color: Colors.transparent, width: 1)),
    minX: null,
    maxX: null,
    minY: 0,
    maxY: 100,
    lineBarsData: [
      LineChartBarData(
        spots: [
          FlSpot.nullSpot,
          FlSpot.nullSpot,
          FlSpot(2, 83),
          FlSpot(3, 83),
          FlSpot(4, 82.25),
          FlSpot(5, 80),
          FlSpot(6, 79),
          FlSpot.nullSpot,
          FlSpot.nullSpot,
          FlSpot.nullSpot,
          FlSpot(10, 79),
        ],
        isCurved: true,
        curveSmoothness: 0.3,
        preventCurveOverShooting: true,
        preventCurveOvershootingThreshold: 0.05,
        colors: [ColorPalette.aquaMarine],
        barWidth: 3,
        isStrokeCapRound: true,
        dotData: FlDotData(
            show: true,
            getDotPainter: (spot, percent, barData, index) => FlDotCirclePainter(color: Colors.white)),
        belowBarData: BarAreaData(
          show: true,
          gradientColorStops: const [0.0, 0.5, 1.0],
          gradientFrom: const Offset(0.5, 1),
          gradientTo: const Offset(0.5, 0),
          colors:
              gradientColors.map((color) => color.withOpacity(0.2)).toList(),
        ),
      ),
    ],
  );
```

**Versions**
Flutter 2.0.3
fl_chart: 0.35.0

**Comment by pulkitjainpsi** [comment]
you can pass first spot as FlSpot(0,0) and  do you able to tap on left and right border spot ?

**Comment by andreferreiravntrs** [comment]
Its possiblei to set the first spot as FlSpot(0,0) but then that spot will be clickable, even if i try to make it invisible

**Comment by andreferreiravntrs** [comment]
I have also tried to check if the dotData was influencing the error... but its all related with the spots itselft

```
        dotData: FlDotData(
            checkToShowDot: (spot, barData) => spot.y <= 0.0 ? false : true,
            getDotPainter: (spot, percent, barData, index) =>
                FlDotCirclePainter(color: Colors.white)),
```

**Comment by imaNNeoFighT** [comment]
Hi.
I'm pretty sure it must be fixed in [0.36.0](https://github.com/imaNNeoFighT/fl_chart/releases/tag/0.36.0)
Can you take a look?
